### PR TITLE
Feature/edit delete resource

### PR DESCRIPTION
**Pull request by kimberlym4488**
Let's test edit/delete a resource...

- [ ] After adding an image, a button should show on the card for delete.
- [ ] After adding a link or markdown, two buttons should show - edit / delete.
- [ ] Edit/Delete buttons show on idea view and in the resource view**
- [ ] Delete - simple client side click event with basic confirm message to make sure you intend to delete. Sorry, I know it's basic ;)
- [ ] Editing an existing markdown/link should show a modal with current content. 
 *  User should be able to edit name and content. 
 *  Submit should render a successful toast message and updated content.

> ** Only idea owners delete/edit resources. Let me know if this should be the resource owner. But if an idea owner can't moderate that could be a problem as well. Maybe it should be both of them could edit/delete?  Just lmk

### CODE REVIEW

File: views/idea.handlebars [hunk 1]
```diff
@@ -125,15 +125,34 @@
...
+                            class='mb-3 btn btn-danger delete-resource'
+                            value='{{id}}'
+                            data-id='{{id}}'
+                        >Delete</button>
+                   
+                        {{#unless (is_match type 'image')}}
+                        <button
+                            class='mb-3 btn btn-success'
+                            value='{{id}}'
+                            type="{{type}}"
```
Comment (jmichaelbrown8) [inline views/idea.handlebars:141]: I think you want a different html attribute here. The button type attribute is usually something like 'button' or 'submit': https://www.w3schools.com/tags/att_button_type.asp

**Comment by kimberlym4488** [comment]
@MikiWolfe would you be free to help with some troubleshooting? Modal is pulling up with data, but it's always the first resource, the click event/ event.currentTarget isn't working. I tried the solution we did with likes (forEach create event listener) but that also wasn't working. Any help is greatly appreciated! Main files to looks at are idea.handlebars lines 114:157 (where the button is stored) and resource.js, lines 125:154.  JS line 130 is probably the issue. 

 The console is showing the  item has the correct id, name, content, etc displayed in the box.
`
//resource.js line 130 is where it keeps grabbing the same id no matter what
  const resource_id = $('.editing-buttons').attr('data-id');
`
`   {{#unless (is_match type 'image')}}
                        <button
                            class='mb-3 btn btn-success text-decoration-none current-resource' 
                            data-bs-target="#edit-{{type}}"
                            data-bs-toggle='modal'
                        >Edit</button> `

File: public/js/resource.js [hunk 2]
```diff
@@ -41,6 +40,123 @@ const linkUpload = () => {
...
+    document.location.href = `/space/${space_id}/idea/${idea_id}`;
+  } else {
+    localStorage.setItem('toast', `Unable to update this :().`);
+    toastIt(true);
+  }
+};
+
+const currentResource = async (event) => {
+  event.preventDefault();
+  event.currentTarget;
```
Comment (jmichaelbrown8) [inline public/js/resource.js:127]: event.currentTarget will give you the exact button the user clicked--if you were to add the resource id to that button, maybe you could use it down below (but you're not saving this as a variable or using it below).

File: public/js/resource.js [hunk 3]
```diff
@@ -41,6 +40,123 @@ const linkUpload = () => {
...
+
+  const { space_id, idea_id } = getContext();
+  const resource_id = $('.editing-buttons').attr('data-id');
```
Comment (jmichaelbrown8) [inline public/js/resource.js:130]: $('.editing-buttons') will return an array of elements that match the editing-buttons class, so by default, jQuery will grab the first one in the list to chain off of--that's why you're getting that first one on the .attr() call afterwards.

Try passing the exact resource id you want with the event target from above.

**Review by jmichaelbrown8 — APPROVED** [review]
Works for me, nice job!

**Comment by jmichaelbrown8** [comment]
I just thought of a couple of additional enhancements, but I think we should merge for now to give people the ability to edit/delete resources:

- [ ] We could remove the image from the server when it's deleted from the resource table with an afterDestroy hook like Anthony did in models/Idea.js:102
- [ ] We could prevent anyone but the idea owner from adding resources (currently it's only the edit/delete that are limited to the idea owner)

**Pull request merged**